src/librustc_codegen_llvm/base.rs

```diff
@@ -642,7 +642,7 @@ fn write_metadata<'a, 'gcx>(tcx: TyCtxt<'a, 'gcx, 'gcx>,
             config::CrateTypeDylib |
             config::CrateTypeProcMacro => MetadataKind::Compressed,
         }
-    }).max().unwrap();
+    }).max().unwrap_or(MetadataKind::None);
```

Comment:
This is not the source of the regression. Reaching this code is the source of the regression. But it seems a sensible change anyway imo.

If preferred I'll do a bisect and figure out where the real regression happened